scripts/prepare_sector_network.py

```diff
@@ -4593,139 +4790,36 @@ def add_industry(
         / industrial_demand.loc[nodes, "naphtha"].sum()
     )
     # link to supply the naphtha for industry load
-    n.add(
-        "Link",
-        spatial.oil.naphtha,
-        bus0=spatial.oil.nodes,
-        bus1=spatial.oil.naphtha,
-        bus2=spatial.co2.process_emissions,
-        carrier="naphtha for industry",
-        p_nom_extendable=True,
-        efficiency2=process_co2_per_naphtha,
-    )
-
-    non_sequestered = 1 - get(
-        cf_industry["HVC_environment_sequestration_fraction"],
-        investment_year,
-    )
-    # energetic efficiency from naphtha to HVC
-    HVC_per_naphtha = (
-        costs.at["oil", "CO2 intensity"] - process_co2_per_naphtha
-    ) / costs.at["oil", "CO2 intensity"]
-
-    # distribute HVC waste across population
-    if len(spatial.oil.non_sequestered_hvc) == 1:
-        HVC_potential = p_set_naphtha.sum() * nhours * non_sequestered * HVC_per_naphtha
+    if options["waste"]:
+        n.add(
+            "Link",
+            spatial.oil.naphtha,
+            bus0=spatial.oil.nodes,
+            bus1=spatial.oil.naphtha,
+            bus2=spatial.co2.process_emissions,
+            carrier="naphtha for industry",
+            p_nom_extendable=True,
+            efficiency2=process_co2_per_naphtha,
+        )
     else:
-        HVC_potential_sum = (
-            p_set_naphtha.sum() * nhours * non_sequestered * HVC_per_naphtha
+        emitted_co2_per_naphtha = (
+            costs.at["oil", "CO2 intensity"] - process_co2_per_naphtha
+        )
+        non_sequestered = 1 - get(
+            cf_industry["HVC_environment_sequestration_fraction"], investment_year
+        )
+        n.add(
+            "Link",
+            spatial.oil.naphtha,
+            bus0=spatial.oil.nodes,
+            bus1=spatial.oil.naphtha,
+            bus2="co2 atmosphere",
+            bus3=spatial.co2.process_emissions,
+            carrier="naphtha for industry",
+            p_nom_extendable=True,
+            efficiency2=emitted_co2_per_naphtha * non_sequestered,
```

Comment:
@fneum I understand the wish to keep existing functionality, but for me one of the major points for this PR was to get rid of this rather obscure part of the model. I would prefer to have the waste emissions where they occur in the real world, i.e. in the waste sector rather than in the industry demand.

If you are worried about additional links: We could configure the default case to be an active waste sector with only one EU load for HVC waste. This would lead to only 1 additional link. Besides, if the right constraints are in place, the setup of generator + link + load is completely determined, and i hope that gurobi would be smart enough to eliminate it during presolve (not sure about that though, could be a good question for the Gurobi workshop in June)